File: kyulee/programmers-2/소수찾기.java

public class 소수찾기 {
    static boolean[] visited;
    static HashSet<Integer> set;

    public static int solution(String numbers) {
        int answer = 0;
        visited = new boolean[numbers.length()];
        set = new HashSet<>();

        for (int i = 1; i <= numbers.length(); i++) {
            permutation(numbers, "", i);
        }

        for (int num : set) {
            if (isPrime(num)) {
                answer++;
            }
        }

        return answer;
    }

    static void permutation(String numbers, String current, int length) {
        if (current.length() == length) {
            set.add(Integer.parseInt(current));
            return;
        }

        for (int i = 0; i < numbers.length(); i++) {
            if (!visited[i]) {
                visited[i] = true;
                permutation(numbers, current + numbers.charAt(i), length);
                visited[i] = false;
            }
        }
    }

    static boolean isPrime(int num) {
        if (num < 2) {
            return false;
        }
        for (int i = 2; i <= Math.sqrt(num); i++) {
            if (num % i == 0) {
                return false;
            }
        }
        return true;
    }
}

File: sanghoon/programmers-2/소수 찾기.py
from itertools import permutations
def solution(numbers):
    answer = 0
    num = set()

    for i in range(1, len(numbers) + 1) :
        num |= set(map(''.join, permutations(numbers, i)))
    num = set(map(int, num))

    for j in num :
        if isprime(j) :
            answer += 1
    print(answer)
    return answer
def isprime(N):
    if N == 0 or N == 1:
        return False
    if N == 2 or N == 3:
        return True
    for i in range(2, int(N ** 0.5) + 1):
        if N % i == 0:
            return False
    return True


numbers = "17"
solution(numbers)